clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -318,6 +325,11 @@ public class KafkaAdminClient extends AdminClient {
     private final Logger log;
     private final LogContext logContext;
 
+    /**
+     * The name of the internal raft metadata topic
+     */
+    private static final String METADATA_TOPIC_NAME = "__cluster_metadata";
```

Comment:
Hmm, I hadn't really been thinking about the fact that we would have to expose this to the client. I guess that is the consequence of having such a general `DescribeQuorum` API. This makes me wonder if we ought to be more forward looking with the naming here. Suppose that we ultimately decide to use raft for partition replication as well. Then we might want to be able to use `DescribeQuorum` for user partitions as well, but we haven't given ourselves a lot of room for extension in the `describeQuorum` API. Would it make sense to 
```java
public DescribeMetadataQuorumResult describeMetadataQuorum(DescribeMetadataQuorumOptions options)
```
It is more verbose, but it is also clearer.

We should also move this constant to `org.apache.kafka.common.internals.Topic`.

Reply:
This is a great point! I think it makes sense to give ourselves wiggle room and call this `MetadataQuorum`.
Re: moving the constant -- I was actually looking for a common place to put it so that both the clients and the server code can use the same value. Is there any class that they share? Right now the name for the topic on the server side is defined in `KafkaRaftServer`.